src/main/java/hudson/plugins/junitattachments/AttachmentPublisher.java

```diff
@@ -33,7 +33,7 @@
     public AttachmentPublisher() {
     }
 
-    public static FilePath getAttachmentPath(AbstractBuild<?, ?> build) {
+    public static FilePath getAttachmentPath(Run<?, ?> build) {
```

Comment:
I do not think binary compatibility is relevant for this plugin, unless you know of some downstream plugin.